tests/fortran/ranges_test.py

```diff
@@ -0,0 +1,778 @@
+from dace.frontend.fortran.fortran_parser import create_singular_sdfg_from_string
+"""
+We test for the following patterns:
+* Range 'ALL'
+* selecting one element by constant
+* selecting one element by variable
+* selecting a subset (proper range) through constants
+* selecting a subset (proper range) through variables
+* ECRAD patterns (WiP)
+  flux_dn(:,1:i_cloud_top) = flux_dn_clear(:,1:i_cloud_top)
+* Extended ECRAD pattern with different loop starting positions.
+* Arrays with offsets
+* Assignment with arrays that have no range expression on the right
+"""
+
+
+def test_fortran_frontend_multiple_ranges_all():
+    """
+    Tests that the generated array map correctly handles offsets.
+    """
+    test_string = """
+                    PROGRAM multiple_ranges
+                    implicit none
+                    double precision, dimension(7) :: input1
+                    double precision, dimension(7) :: input2
+                    double precision, dimension(7) :: res
+                    CALL multiple_ranges_function(input1, input2, res)
+                    end
+
+                    SUBROUTINE multiple_ranges_function(input1, input2, res)
+                    double precision, dimension(7) :: input1
+                    double precision, dimension(7) :: input2
+                    double precision, dimension(7) :: res
+
+                    res(:) = input1(:) - input2(:)
+
+                    END SUBROUTINE multiple_ranges_function
+                    """
+
+    sdfg = fortran_parser.create_sdfg_from_string(test_string, "multiple_ranges", True)
+    sdfg.simplify()
+    sdfg.compile()
+
+    size = 7
+    input1 = np.full([size], 0, order="F", dtype=np.float64)
+    input2 = np.full([size], 0, order="F", dtype=np.float64)
+    for i in range(size):
+        input1[i] = i + 1
+        input2[i] = i
+    res = np.full([7], 42, order="F", dtype=np.float64)
+    sdfg(input1=input1, input2=input2, res=res)
+    for val in res:
+        assert val == 1.0
+
+
+def test_fortran_frontend_multiple_ranges_selection():
+    """
+    Tests that the generated array map correctly handles offsets.
+    """
+    test_string = """
+                    PROGRAM multiple_ranges_selection
+                    implicit none
+                    double precision, dimension(7,2) :: input1
+                    double precision, dimension(7) :: res
+                    CALL multiple_ranges_selection_function(input1, res)
+                    end
+
+                    SUBROUTINE multiple_ranges_selection_function(input1, res)
+                    double precision, dimension(7,2) :: input1
+                    double precision, dimension(7) :: res
+
+                    res(:) = input1(:, 1) - input1(:, 2)
+
+                    END SUBROUTINE multiple_ranges_selection_function
+                    """
+
+    sdfg = fortran_parser.create_sdfg_from_string(test_string, "multiple_ranges_selection", True)
+    #sdfg.simplify()
```

Comment:
why can simplify be disabled on a two-dimensional array? Don't you need scal2sym?